cmd/docker-mcp/internal/gateway/handlers.go

```diff
@@ -26,6 +51,43 @@ func (g *Gateway) mcpToolHandler(tool catalog.Tool) mcp.ToolHandler {
 
 func (g *Gateway) mcpServerToolHandler(serverConfig *catalog.ServerConfig, server *mcp.Server, annotations *mcp.ToolAnnotations) mcp.ToolHandler {
 	return func(ctx context.Context, ss *mcp.ServerSession, params *mcp.CallToolParamsFor[map[string]any]) (*mcp.CallToolResultFor[any], error) {
+		// Debug logging to stderr
+		if os.Getenv("DOCKER_MCP_TELEMETRY_DEBUG") != "" {
+			fmt.Fprintf(os.Stderr, "[MCP-HANDLER] Tool call received: %s from server: %s\n", params.Name, serverConfig.Name)
+		}
+		
+		// Start telemetry span for tool call
+		startTime := time.Now()
+		serverType := inferServerType(serverConfig)
+		
+		// Build span attributes
+		spanAttrs := []attribute.KeyValue{
+			attribute.String("mcp.server.name", serverConfig.Name),
```

Comment:
Not for today, but I think this could become an oci ref that would allow us to index a lot of rich information about the server. This name is currenty an "alias" within a particular version of the catalog and without knowing that version, we'll obviously be missing a lot of information about the identity of the server.  So I guess my only point here is that I think we'll want to update this value to make it more useful.